### DISCUSSION TITLE
How to fully reset MetaMask between tests?

### DISCUSSION THREAD

**Discussion by sebastiendan** [Synthetixio/synpress#769] [Q&A]
Hi,

I'm interested in the cleanest way to reset MetaMask's state between tests. We have the `disconnectMetamaskWalletFromAllDapps` and `resetMetamaskAccount` methods but I couldn't find anything for networks.

Because of this, tests cannot be fully isolated and stateless when one's switching networks, i.e., the new network will remain active for the next test.

**Comment by r3kt-eth** [comment]
It's possible but has to be introduced to synpress. Here's a mostly working solution:
```
clearExtensionData: async extensionWelcomeUrl => {
    await metamaskWindow.evaluate(async () => {
      chrome.storage.local.clear();
      chrome.storage.sync.clear();
      chrome.runtime.reload();
    });
    await mainWindow.waitForTimeout(1000);
    const newPagePromise = new Promise(resolve =>
      browser.contexts()[0].once('page', resolve),
    );
    await mainWindow.evaluate(async extensionWelcomeUrl => {
      window.open(extensionWelcomeUrl, '_blank').focus();
    }, extensionWelcomeUrl);
    metamaskWindow = await newPagePromise;
    await module.exports.assignActiveTabName('metamask');
    await metamaskWindow.reload();
    await module.exports.waitUntilStable();
    return metamaskWindow;
  },
```

**Reply by sebastiendan** [reply]
Hey @r3kt-eth, would you have more context to share as to where this should be set?
